configure.ac

```diff
@@ -701,9 +701,28 @@ if test "$with_skey" = "yes"; then
 	]])],[AC_DEFINE(SKEY_BSD_STYLE, 1, [Define to support newer BSD S/Key API])],[])
 fi
 
-PKG_CHECK_MODULES([CMOCKA], [cmocka], [have_cmocka="yes"],
-	[AC_MSG_WARN([libcmocka not found, cmocka tests will not be built])])
+AC_COMPILE_IFELSE(
+                  [AC_LANG_SOURCE([
+                      #include <stdarg.h>
+                      #include <stddef.h>
+                      #include <setjmp.h>
+                      #include <stdint.h>
+                      #include <cmocka.h>
+
+                      int main(void)
+                      {
+                          assert_int_equal(1, 1);
+                      }
+                  ])],
+                  [
+                      [have_cmocka="yes"]
+                      [CMOCKA_LIBS="-lcmocka"]
+                  ],
+                  [
+                      AC_MSG_WARN([libcmocka not found, cmocka tests will not be built])
+                  ])
 AM_CONDITIONAL([HAVE_CMOCKA], [test x$have_cmocka = xyes])
+AC_SUBST(CMOCKA_LIBS)
```

Comment:
I can't say I reviewed it, because I don't understand it either.  :-)

But I trust you know what you're doing, so if nobody complains in a reasonable time, feel free to merge.  As long as you test it and it works, it's ok for me.

Maybe @hallyn knows.